app/services/publication-service.js

```diff
@@ -111,4 +111,31 @@ export default class PublicationService extends Service {
   invalidatePublicationCache() {
     this.cachedData = A([]);
   }
+
+  async createNumacNumber(name) {
+    const numacNumber = this.store.createRecord('numac-number', {
+      name: name,
+    });
+    return await numacNumber.save();
+  }
+
+  async linkNumacNumber(numacNumber, publicationFlow) {
+    const numberslist = await publicationFlow.get('numacNumbers');
+    numberslist.pushObject(numacNumber);
+    publicationFlow.set('numacNumbers', numberslist);
+    return await publicationFlow.save();
+  }
+
+  async unlinkNumacNumber(numacNumber, publicationFlow) {
+    console.log(numacNumber);
+    const numberslist = await publicationFlow.get('numacNumbers').toArray();
+    for (let index = 0; index < numberslist.length; index++) {
+      const numacNumberInLoop = numberslist[index];
+      if (numacNumberInLoop.get('id') === numacNumber.get('id')) {
+        numberslist.splice(index, 1);
+      }
+    }
+    publicationFlow.set('numacNumbers', numberslist);
+    return await publicationFlow.save();
+  }
```

Comment:
dit is 1 aanpak, wat je ook kan doen is numacNummer een publication-flow meegeven.
Dus de belongsTo kant updaten van de one-to-many.  (betere manier ?)
En daarna de relatie hasmany('numacNumbers') reloaden

Zelfde met unlinken, maar kan zelfs gemakkelijker.
Gewoon die numac-nummer deleteRecord doen
daarna de relatie hasmany('numacNumbers') reloaden
Dat hele splicen en dan een nieuwe lijst setten is wat omslachtig

Want wat je nu niet doet is dat model van numac-number opkuisen, dat bestaat nog in data

Reply:
Aangepast naar deze manier. Is inderdaad eleganter.

Reply:
Ik heb hiervoor ook wel de backend PR wat moeten aanpassen (die lag nog niet in 2 richtingen, nu dus wel)